src/js/components/nft/nft-collection-view.jsx

```diff
@@ -0,0 +1,258 @@
+import React, {Component} from 'react';
+import {withRouter} from 'react-router-dom';
+import {Button, Card, Col, Form, Row} from 'react-bootstrap';
+import API from '../../api';
+import PhotoAlbum from 'react-photo-album';
+import {connect} from 'react-redux';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import * as format from '../../helper/format';
+import * as svg from '../../helper/svg';
+import ModalView from '../utils/modal-view';
+import * as text from '../../helper/text';
+import {changeLoaderState} from '../loader';
+import Transaction from '../../common/transaction';
+import async from 'async';
+import utils from '../../helper/utils';
+
+
+class NftCollectionView extends Component {
+    constructor(props) {
+        super(props);
+        this.datatable_reload_interval = undefined;
+        this.state                     = {
+            nft_list                    : [],
+            nft_selected                : undefined,
+            datatable_reload_timestamp  : '',
+            datatable_loading           : false,
+            modal_show_burn_confirmation: false,
+            modal_show_burn_result      : false,
+            modal_burn_create_asset     : true,
+            modal_body_burn_result      : []
+        };
+    }
+
+    componentDidMount() {
+        this.reloadCollection();
+        this.datatable_reload_interval = setInterval(() => this.reloadCollection(), 60000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.datatable_reload_interval);
+        this.state.nft_list.forEach(nft => nft.src && URL.revokeObjectURL(nft.src));
+    }
+
+    reloadCollection() {
+        this.setState({
+            datatable_loading: true
+        });
+
+        return API.listTransactionWithDataReceived(this.props.wallet.address_key_identifier, 'tangled_nft').then(data => {
+            async.mapLimit(data, 6, (row, callback) => {
+                utils.getImageFromApi(row)
+                    .then(imageUrl => callback(null, {
```

Comment:
callbacks that you use after getImageFromApi which is used after listTransactionWithDataReceived looks extremely similar. probably it worth to add a helper and put handling there. or add it as a part of getImageFromApi func